exif-APP12-SOFn-infoleak: exif: out of bounds read in the APP12 Info string and in a short SOFn JPEG section.
Upstream https://github.com/php/php-src/commit/e59b6dc0ae80

diff --git a/ext/exif/exif.c b/ext/exif/exif.c
index 5843577..26fad9c 100644
--- a/ext/exif/exif.c
+++ b/ext/exif/exif.c
@@ -3145,7 +3145,7 @@ static void exif_process_APP12(image_info_type *ImageInfo, char *buffer, size_t
 	if ((l1 = php_strnlen(buffer+2, length-2)) > 0) {
 		exif_iif_add_tag(ImageInfo, SECTION_APP12, "Company", TAG_NONE, TAG_FMT_STRING, l1, buffer+2 TSRMLS_CC);
 		if (length > 2+l1+1) {
-			l2 = php_strnlen(buffer+2+l1+1, length-2-l1+1);
+			l2 = php_strnlen(buffer+2+l1+1, length-2-l1-1);
 			exif_iif_add_tag(ImageInfo, SECTION_APP12, "Info", TAG_NONE, TAG_FMT_STRING, l2, buffer+2+l1+1 TSRMLS_CC);
 		}
 	}
@@ -3291,6 +3291,10 @@ static int exif_scan_JPEG_header(image_info_type *ImageInfo TSRMLS_DC)
 			case M_SOF13:
 			case M_SOF14:
 			case M_SOF15:
+				if ((itemlen - 2) < 6) {
+					return FALSE;
+				}
+		
 				exif_process_SOFn(Data, marker, &sof_info);
 				ImageInfo->Width  = sof_info.width;
 				ImageInfo->Height = sof_info.height;
